htmresearch/frameworks/nlp/classify_htm.py

```diff
@@ -418,27 +171,39 @@ def trainText(self, token, labels, sequenceId=None, reset=0):
       self.reset()
 
 
-  def classifyText(self, token, reset=0):
+  def inferToken(self, token, reset=0, returnDetailedResults=False,
+                 sortResults=True):
     """
-    Classify the token and return a list of the best classifications.
+    Classify the token (i.e. run inference on the model with this document) and
+    return classification results and a list of sampleIds and distances.
+    Repeated sampleIds are NOT removed from the results.
 
-    @param token    (str)  The text token to train on
-    @param reset    (int)  Should be 0 or 1. If 1, assumes we are at the
-                           end of a sequence. A reset signal will be issued
-                           after the model has been trained on this token.
+    @param token    (str)     The text token to train on
+    @param reset    (int)     Should be 0 or 1. If 1, assumes we are at the
+                              end of a sequence. A reset signal will be issued
+                              after the model has been trained on this token.
+    @param sortResults (bool) If true the list of sampleIds and distances
+                              will be sorted in order of increasing distances.
 
     @return  (numpy array) An array of size numLabels. Position i contains
-                           the likelihood that this sample belongs to the
+                           the likelihood that this token belongs to the
                            i'th category. An array containing all zeros
                            implies no decision could be made.
+             (list)        A list of sampleIds
+             (numpy array) An array of distances from each stored sample
     """
     for region in self.learningRegions:
       region.setParameter("learningMode", False)
       region.setParameter("inferenceMode", True)
     sensor = self.sensorRegion.getSelf()
-    sensor.addDataToQueue(token, [None], -1, 0)
+    sensor.addDataToQueue(token, [None], sequenceId=-1, reset=0)
     self.network.run(1)
 
+    dist = self.classifierRegion.getSelf().getLatestDistances()
+
+    categoryLikelihoods = self.classifierRegion.getOutputData(
+        "categoriesOut")[0:self.numLabels]
```

Comment:
see comment in htmresearch/frameworks/nlp/classify_document_fingerprint.py about this use of `numLabels`